rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/parametric/InterpretedLanguageContributions.java

```diff
@@ -206,13 +211,26 @@ private static ISet loadContributions(Evaluator eval, LanguageParameter lang) {
         return (ISet) eval.eval(eval.getMonitor(), lang.getMainFunction() + "()", URIUtil.rootLocation("lsp"))
             .getValue();
     }
+ 
+    @Override
+    public CompletableFuture<IList> parseCodeActions(String command) {
+        return store.thenApply(commandStore -> {
+            try {
+                var TF = TypeFactory.getInstance();
+                return (IList) new StandardTextReader().read(VF, commandStore, TF.listType(commandStore.lookupAbstractDataType("CodeAction")), new StringReader(command));
+            } catch (FactTypeUseException | IOException e) {
+                logger.catching(e);
```

Comment:
I'm pretty sure these exceptions are impossible to happen. Since we write the data and then read it back in. But in any case it can't do harm to communicate then you the user.